### PR TITLE
Add type-promotion for cat's mapping to oneDNN Graph

### PR DESCRIPTION
**Pull request by sanchitintel**
## Bug
Now that the oneDNN Graph fuser supports BFloat16 dtype as well, type-promotion needs to be handled.
Some of `aten::cat`'s tensor inputs should undergo type promotion, if they're BF16 while at least one input is FP32.
oneDNN Graph would throw a runtime error if all inputs of the `concat` op are not of the same dtype.

## Fix
If the dtype of one of the inputs is FP32, the the output would be FP32. So the non-FP32 inputs should be promoted to FP32 as well.

## Verification method
Added a UT - `test_cat_check_type_promotion`

### CODE REVIEW

File: test/test_jit_llga_fuser.py [hunk 1]
```diff
@@ -366,6 +366,23 @@ def forward_cat(*inputs):
...
+                return torch.cat(inputs, d)
+            return forward_cat
+
+        for xshape, yshape, dim in [
+            [[4, 8, 8, 16], [4, 8, 8, 24], 3],
+            [[64, 8, 32], [64, 24, 32], 1],
+            [[2048, 64], [1024, 64], 0],
+        ]:
+            x = torch.rand(xshape)
+            y = torch.rand(yshape, dtype=torch.bfloat16)
```
Comment (jgong5) [inline test/test_jit_llga_fuser.py:382]: This won't happen if users do autocast, right? I guess autocast would promote types for `cat`.

**Review by jgong5 — CHANGES_REQUESTED** [review]

File: test/test_jit_llga_fuser.py (hunk 1 above)
Comment (sanchitintel) [reply test/test_jit_llga_fuser.py:382]: Oh, maybe that's why we didn't need this commit 25 days back. A commit changed some AMP functionality 22 days back. I'll dig deeper.

Comment (sanchitintel) [reply test/test_jit_llga_fuser.py:382]: I had thought we didn't support AMP for `cat`, so I opened this PR.

`cat`'s AMP implementation seems to be broken for eager mode on CPU. So, I'll close this PR, as it's redundant. Thanks!

**Comment by sanchitintel** [comment]
I had thought we didn't support AMP for `cat`, so I opened this PR.

`cat`'s AMP implementation seems to be broken for eager mode on CPU. So, I'll close this PR, as it's redundant. Thanks!

**Pull request closed without merging**